core/services/pipeline/task.hexencode.go

```diff
@@ -0,0 +1,72 @@
+package pipeline
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+//
+// Return types:
+//     bytes
+//
+type HexEncodeTask struct {
+	BaseTask `mapstructure:",squash"`
+	Input    string `json:"input"`
+}
+
+var _ Task = (*HexEncodeTask)(nil)
+
+func (t *HexEncodeTask) Type() TaskType {
+	return TaskTypeHexEncode
+}
+
+func addHexPrefix(val string) string {
+	if len(val) > 0 {
+		return "0x" + val
+	} else {
+		return ""
+	}
+}
+
+func (t *HexEncodeTask) Run(_ context.Context, _ logger.Logger, vars Vars, inputs []Result) (result Result, runInfo RunInfo) {
+	_, err := CheckInputs(inputs, 1, 1, 0)
+	if err != nil {
+		return Result{Error: errors.Wrap(err, "task inputs")}, runInfo
+	}
+
+	if strValue, ok := inputs[0].Value.(string); ok {
```

Comment:
I think this approach is wrong, because it will not support expressions like `$(other_task.result)`.
In other tasks you could find statements like this:
`errors.Wrap(ResolveParam(&input, From(VarExpr(t.Input, vars), Input(inputs, 0))), "input"),`
Here, `VarExpr` is handling the case of expressions. However, this requires to know the exact variable type.
At the same time we do not support types like "variant".
I would suggest to limit the supported types to:
- string
- []byte
- uint/int of arbitrary precision
Then you can do a chain of if-else for `errors.Wrap(ResolveParam(&input, From(VarExpr(t.Input, vars), Input(inputs, 0))), "input"),` trying to match the parameter type, and failing in the last `else`.
Would this work?

Other approach is to introduce a new param type, like `VariantParam` and then do what `ETHABIEncodeTask2` does in `convertToETHABIType` function.